hazelcast-client-new/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -132,6 +133,7 @@ protected Document parse(InputStream inputStream) throws Exception {
         DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
         dbf.setNamespaceAware(true);
         final DocumentBuilder builder = dbf.newDocumentBuilder();
+        Locale.setDefault(Locale.getDefault());
```

Comment:
I think JVM is already using default locale if locale not provided.. check [toUpperCase](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#toUpperCase())

Instead of this PR, we should add a new complete feature like overriding default locale if there is such need...